cli/src/semgrep/metrics.py

```diff
@@ -249,6 +249,10 @@ def add_rules(self, rules: Sequence[Rule], profile: Optional[out.Profile]) -> No
                     bytesScanned=mock_int(_rule_bytes_scanned[rule.id2]),
                 )
                 for rule in rules
+                # We consider only rules with match times and bytes scanned
+                # greater than 0 to avoid making the metrics too bloated.
+                if _rule_match_times[rule.id2] > 0.0
```

Comment:
maybe would be good to have some tests for that (the time will differ between runs, but we have some code
in pytest to mask times.
That way we can also mark this tests as 'osemfail' so we know we need to port it.

Reply:
Good idea. I'll try to add that change before merging.

Reply:
But first, I probably need to figure out how to trigger this code path in pysemgrep.

Reply:
Done. I was able to piggyback on one of the tests that test the metrics payload. Waiting for CI to merge.